src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepTest.java

```diff
@@ -0,0 +1,71 @@
+        step.setName("podTest");
+        st.assertRoundTrip(step, "podTemplate(name: 'podTest') {\n    // some block\n}");
+        step.setName("");
+        step.setInstanceCap(5);
+        st.assertRoundTrip(step, "podTemplate(instanceCap: 5) {\n    // some block\n}");
+        step.setInstanceCap(0);
+        step.setInstanceCap(6);
+        st.assertRoundTrip(step, "podTemplate(instanceCap: 6) {\n    // some block\n}");
+        step.setInstanceCap(null); // make sure this resets instanceCap
+        st.assertRoundTrip(step, "podTemplate {\n    // some block\n}");
+        step.setInstanceCap(7);
+        st.assertRoundTrip(step, "podTemplate(instanceCap: 7) {\n    // some block\n}");
+        step.setInstanceCap(Integer.MAX_VALUE); // make sure this resets instanceCap
+        st.assertRoundTrip(step, "podTemplate {\n    // some block\n}");
+        step.setLabel("podLabel");
+        st.assertRoundTrip(step, "podTemplate(label: 'podLabel') {\n    // some block\n}");
+        step.setLabel("");
+        st.assertRoundTrip(step, "podTemplate {\n    // some block\n}");
+        step.setPodRetention(PodRetention.getPodTemplateDefault()); // this is the default, it should not appear in the snippet.
+        st.assertRoundTrip(step, "podTemplate {\n    // some block\n}");
+        step.setPodRetention(new OnFailure());
+        st.assertRoundTrip(step, "podTemplate(podRetention: onFailure()) {\n    // some block\n}");
+        step.setPodRetention(null);
+        st.assertRoundTrip(step, "podTemplate {\n    // some block\n}");
+        step.setWorkspaceVolume(new DynamicPVCWorkspaceVolume());
+        st.assertRoundTrip(step, "podTemplate(workspaceVolume: dynamicPVC()) {\n    // some block\n}");
+        step.setWorkspaceVolume(new EmptyDirWorkspaceVolume(false)); // this is the default, it should not be in the snippet.
+        st.assertRoundTrip(step, "podTemplate {\n    // some block\n}");
+        step.setWorkspaceVolume(new DynamicPVCWorkspaceVolume(null, null, "ReadWriteMany"));
+        st.assertRoundTrip(step, "podTemplate(workspaceVolume: dynamicPVC(accessModes: 'ReadWriteMany')) {\n    // some block\n}");
+        step.setWorkspaceVolume(null);
+        st.assertRoundTrip(step, "podTemplate {\n    // some block\n}");
+        step.setActiveDeadlineSeconds(60);
+        st.assertRoundTrip(step, "podTemplate(activeDeadlineSeconds: 60) {\n    // some block\n}");
+        step.setActiveDeadlineSeconds(0);
+        st.assertRoundTrip(step, "podTemplate {\n    // some block\n}");
+        step.setInheritFrom("fooBar");
+        st.assertRoundTrip(step, "podTemplate(inheritFrom: 'fooBar') {\n    // some block\n}");
+        step.setInheritFrom(",default,");
+        assertEquals("inheritFrom not changed from \",default,\" to \"\" by setter", "", step.getInheritFrom());
+        step.setInheritFrom(null);
+        st.assertRoundTrip(step, "podTemplate {\n    // some block\n}");
```

Comment:
And then you also need something like
```suggestion
        st.assertRoundTrip(step, "podTemplate {\n    // some block\n}");
        step.setInheritFrom("");
        st.assertRoundTrip(step, "podTemplate(inheritFrom: '') {\n    // some block\n}");
```

Reply:
Covered through https://github.com/jenkinsci/kubernetes-plugin/pull/703/commits/7309343a0a41fed55a0e9926a977ca58b4b6289c#diff-502362a2cf01135e38702358449886d5R65-R66